app/assets/javascripts/SourceCodeGlower/ImageAnnotationGrid.js

```diff
@@ -116,22 +116,26 @@ ImageAnnotationGrid.prototype.add_to_grid = function(extracted_coords) {
   this.draw_holders();
 }
 
-ImageAnnotationGrid.prototype.remove_annotation = function(unused_param1, unused_param2, annotation_text_id) {
+ImageAnnotationGrid.prototype.remove_annotation = function(annotation_id, unused_param2, annotation_text_id) {
   if (this.getAnnotationTextManager().annotationTextExists(annotation_text_id)) {
-    this.getAnnotationTextManager().removeAnnotationText(annotation_text_id);
-    var holder = document.getElementById('annotation_holder_' + annotation_text_id);
+    var holder = document.getElementById('annotation_holder_' + annotation_id);
     if (holder != null) {
       holder.parentElement.removeChild(holder);
     }
   }
 
   var annot_grid = this.get_annotation_grid();
+  var otherAnnotationsWithText = false;
   for (var i = 0; i < annot_grid.length; i++) {
-    if (annot_grid[i].id == annotation_text_id) {
+    if (annot_grid[i].annot_id == annotation_id) {
       annot_grid.splice(i, 1);
-      break;
+    }else if(annot_grid[i].id == annotation_text_id){
```

Comment:
You're right, please change the one above as well.

We generally want to use `===` all throughout the code, since it is stricter (doesn't implicitly convert types). You can read more about this [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) if you like.